src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprAllRecipes.java

```diff
@@ -0,0 +1,170 @@
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Name("All Recipes")
+@Description({
+	"Retrieve all recipes registered in the server. Includes the options to:",
+	"<ul>",
+	"<li>Get only recipes provided by Minecraft or custom recipes</li>",
+	"<li>Specific recipe types</li>",
+	"<li>For specific items</li>",
+	"</ul>"
+})
+@Examples({
+	"set {_list::*} to all of the server's recipe of type shaped recipe for netherite ingot",
+	"set {_list::*} to all server's mc recipes of type cooking recipe for raw beef",
+	"set {_list::*} to all of the server's custom recipes of type blasting for raw iron named \"Impure Iron\"",
+	"",
+	"reset all of the server's recipes",
+	"reset all recipes for netherite ingot",
+	"reset all of the minecraft recipes",
+	"reset all of the custom shaped recipes"
+})
+@Since("INSERT VERSION")
+public class ExprAllRecipes extends SimpleExpression<Recipe> {
+
+	static {
+		Skript.registerExpression(ExprAllRecipes.class, Recipe.class, ExpressionType.SIMPLE,
+			"[all [of]] [the] server['s] recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] (mc|minecraft|vanilla) recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] custom recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] (mc|minecraft|vanilla) %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] custom %recipetype% [for %-itemstacks/itemtypes%]");
+	}
+
+	private Expression<RecipeType> recipeTypeExpr;
+	private Expression<?> itemExpr;
+	private boolean getMinecraft, getCustom;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern <= 2 && exprs[0] != null) {
+			itemExpr = exprs[0];
+		} else if (matchedPattern >= 3) {
+			//noinspection unchecked
+			recipeTypeExpr = (Expression<RecipeType>) exprs[0];
+			itemExpr = exprs[1] != null ? exprs[1] : null;
+		}
+		getMinecraft = matchedPattern == 1 || matchedPattern == 4;
+		getCustom = matchedPattern == 2 || matchedPattern == 5;
+		return true;
+	}
+
+	@Override
+	protected Recipe @Nullable [] get(Event event) {
+		return getSelectedRecipes(event).toArray(new Recipe[0]);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.RESET)
+			return CollectionUtils.array(Recipe.class);
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (itemExpr == null && recipeTypeExpr == null && !getMinecraft && !getCustom) {
+			Bukkit.resetRecipes();
+		} else {
+			getSelectedRecipes(event).forEach(recipe -> {
+				if (recipe instanceof Keyed key)
+					Bukkit.removeRecipe(key.getKey());
+			});
+		}
+	}
+
+	private List<Recipe> getSelectedRecipes(Event event) {
```

Comment:
I can see how this would be nicer, and I could use it for the `change` method, but what about for the `get`?
Just wanting to get your thoughts, I could still  do it if you want, just have to change the get method

Reply:
you ust collect the iterator into an array